File: 02-week-2--recursion/02-day-2--recursive-search/javascript/recursive_search.js
function recursiveSearch(arr, target) {
  // type your code here
}
function binary_search(arr, target){
  if (arr.length == 0){
    return false
  }
  const mid = Math.floor(arr.length / 2)
  if(arr[mid] == target){
    return true
  }
  else if (arr[mid] < target){
    return binary_search(arr.slice(mid+1), target)
  }
  else {
    return binary_search(arr.slice(0,mid),target)
  }
    
}
if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: true");
  console.log("=>", recursiveSearch([1, 2, 3], 2));

  console.log("");

  console.log("Expecting: false");
  console.log("=>", recursiveSearch([3, 2, 1], 4));
}

module.exports = recursiveSearch;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 02-week-2--recursion/02-day-2--recursive-search/ruby/recursive_search.rb
def recursive_search(arr, target)
  i = 0 || i + 1
   return nil if i >= arr.length

  return arr[i] if arr[i]=target

 recursive_search(arr, target)

end
# first try

def recursively_search(arr, target)
  return false if arr.empty?
  return false if !arr.include?(target)
  recursively_search(arr[0..-1], target)
end
def binary_search(arr, target)
  return false if arr.empty?

  mid = arr.length / 2
  if arr[mid] == target
    return true
  elsif arr[mid] > target
    binary_search(arr[0..mid], target)
  else
    binary_search(arr[(mid+1)..-1], target)
  end
end
puts binary_search([1,2,3,4,5,6,7],7)
if __FILE__ == $PROGRAM_NAME
  puts "Expecting: true"
  puts "=>", recursive_search([1, 2, 3], 2)

  puts

  puts "Expecting: false"
  puts "=>", recursive_search([3, 2, 1], 4)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
